### PR TITLE
Fix `View File` button link if branch deleted on pull request files pages

### PR DESCRIPTION
**Pull request by metiftikci**
as title

### CODE REVIEW

**Review by delvh — APPROVED** [review]

File: routers/web/repo/pull.go [hunk 1]
```diff
@@ -862,7 +862,7 @@ func viewPullFiles(ctx *context.Context, specifiedStartCommit, specifiedEndCommi
 		}
 
 		if pull.HeadRepo != nil {
-			ctx.Data["SourcePath"] = pull.HeadRepo.Link() + "/src/branch/" + util.PathEscapeSegments(pull.HeadBranch)
+			ctx.Data["SourcePath"] = pull.HeadRepo.Link() + "/src/commit/" + endCommitID
```
Comment (lunny) [inline routers/web/repo/pull.go:865]: They are different. `endCommitID` is only the last commit of this pull request.

**Review by lunny — CHANGES_REQUESTED** [review]
per https://github.com/go-gitea/gitea/pull/31063/files#r1612526557

**Comment by metiftikci** [comment]
this is also usefull when reviwing specific commit in pull request

github also not redirecting to branch

so if you do not want to change branch link behaviour, do you have any suggestion ? like should we check if branch deleted ?

**Comment by lunny** [comment]
I mean if there are serveal commits on a pull request and when view files on one commit mode. The link should be that commit's id but not end commit id?

**Comment by metiftikci** [comment]
This is how it work after change:

### Show All Commit: last commit

image

### View Range: last selected commit

image

### View One Commit: selected commit

image

**Review by lunny — APPROVED** [review]

**Pull request merged**